sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/ResourceLabels.java

```diff
@@ -0,0 +1,64 @@
+package com.mesosphere.sdk.offer.evaluate;
+
+import com.mesosphere.sdk.specification.ResourceSpec;
+import org.apache.commons.lang3.builder.ToStringBuilder;
+
+import java.util.Optional;
+
+/**
+ * Pairs a {@link ResourceSpec} definition with an existing task's labels associated with that resource.
+ */
+class ResourceLabels {
+    private final ResourceSpec original;
+    private final ResourceSpec updated;
+
+    private final String resourceId;
+    private final Optional<String> persistenceId;
+    private final Optional<String> sourceRoot;
+
+    public ResourceLabels(ResourceSpec resourceSpec, String resourceId) {
+        this(resourceSpec, resourceSpec, resourceId);
+    }
+
+    public ResourceLabels(ResourceSpec original, ResourceSpec updated, String resourceId) {
+        this(original, updated, resourceId, Optional.empty(), Optional.empty());
+    }
+
+    public ResourceLabels(
```

Comment:
IMO could stand to have some javadocs around `original` vs `updated`.

Reply:
yep.  I'll add some.